### PR TITLE
MotionBlur and geometry buffer renderer to webgpu

### PR DESCRIPTION
**Pull request by deltakosh**

### CODE REVIEW

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Comment by bjsplat** [comment]
Snapshot stored with reference name:
refs/pull/15412/merge

Test environment: 
https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15412/merge/index.html

To test a playground add it to the URL, for example:

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15412/merge/index.html#WGZLGJ#4600

Links to test babylon tools with this snapshot:

https://playground.babylonjs.com/?snapshot=refs/pull/15412/merge
https://sandbox.babylonjs.com/?snapshot=refs/pull/15412/merge
https://gui.babylonjs.com/?snapshot=refs/pull/15412/merge
https://nme.babylonjs.com/?snapshot=refs/pull/15412/merge

To test the snapshot in the playground with a playground ID add it after the snapshot query string:

https://playground.babylonjs.com/?snapshot=refs/pull/15412/merge#BCU1XR#0

**Comment by bjsplat** [comment]
WebGL2 visualization test reporter:

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15412/merge/testResults/webgl2playwright/index.html

**Comment by bjsplat** [comment]
Visualization tests for WebGPU (Experimental)

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15412/merge/testResults/webgpuplaywright/index.html

File: packages/dev/core/src/ShadersWGSL/geometry.fragment.fx [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+    #endif
+
+    var normalOutput: vec3f;
+    #ifdef BUMP
+        var normalW: vec3f = normalize(input.vNormalW);
+        #include<bumpFragment>
+        #ifdef NORMAL_WORLDSPACE
+            normalOutput = normalW;
+        #else
+            normalOutput = normalize( vec3f(vWorldView *  vec4f(normalW, 0.0)));
```
Comment (noname0310) [inline packages/dev/core/src/ShadersWGSL/geometry.fragment.fx:82]:
```suggestion
            normalOutput = normalize( vec3f(input.vWorldView *  vec4f(normalW, 0.0)));

```

File: packages/dev/core/src/ShadersWGSL/geometry.fragment.fx [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+        #ifdef PREPASS_NORMAL
+            fragData[NORMAL_INDEX] =  vec4f(normalOutput, 1.0);
+        #endif
+    #else
+        fragData[0] =  vec4f(input.vViewPos.z / input.vViewPos.w, 0.0, 0.0, 1.0);
+        fragData[1] =  vec4f(normalOutput, 1.0);
+    #endif
+
+    #ifdef POSITION
+        fragData[POSITION_INDEX] =  vec4f(vPositionW, 1.0);
```
Comment (noname0310) [inline packages/dev/core/src/ShadersWGSL/geometry.fragment.fx:107]:
```suggestion
        fragData[POSITION_INDEX] =  vec4f(input.vPositionW, 1.0);
```

File: packages/dev/core/src/ShadersWGSL/geometry.vertex.fx [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+	#endif
+
+	vertexOutputs.position = scene.viewProjection * finalWorld *  vec4f(positionUpdated, 1.0);
+
+	#include<clipPlaneVertex>
+
+	#ifdef NEED_UV
+		#ifdef UV1
+			#if defined(ALPHATEST) && defined(ALPHATEST_UV1)
+			vertexOutputs.vUV = (diffuseMatrix *  vec4f(uvUpdated, 1.0, 0.0)).xy;
```
Comment (noname0310) [inline packages/dev/core/src/ShadersWGSL/geometry.vertex.fx:149]:
```suggestion
			vertexOutputs.vUV = (uniforms.diffuseMatrix *  vec4f(uvUpdated, 1.0, 0.0)).xy;
```

File: packages/dev/core/src/ShadersWGSL/geometry.vertex.fx [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+			#ifdef REFLECTIVITY_UV1
+			vertexOutputs.vReflectivityUV = (uniforms.reflectivityMatrix *  vec4f(uvUpdated, 1.0, 0.0)).xy;
+			#endif
+			#ifdef ALBEDO_UV1
+			vertexOutputs.vAlbedoUV = (uniforms.albedoMatrix *  vec4f(uvUpdated, 1.0, 0.0)).xy;
+			#endif
+		#endif
+		#ifdef UV2
+			#if defined(ALPHATEST) && defined(ALPHATEST_UV2)
+			vertexOutputs.vUV = (diffuseMatrix *  vec4f(uv2, 1.0, 0.0)).xy;
```
Comment (noname0310) [inline packages/dev/core/src/ShadersWGSL/geometry.vertex.fx:166]:
```suggestion
			vertexOutputs.vUV = (uniforms.diffuseMatrix *  vec4f(input.uv2, 1.0, 0.0)).xy;
```

File: packages/dev/core/src/ShadersWGSL/geometry.vertex.fx [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+			#else
+			vertexOutputs.vUV = input.uv2;
+			#endif
+
+			#ifdef BUMP_UV2
+			vertexOutputs.vBumpUV = (uniforms.bumpMatrix *  vec4f(uv2, 1.0, 0.0)).xy;
```
Comment (noname0310) [inline packages/dev/core/src/ShadersWGSL/geometry.vertex.fx:172]:
```suggestion
			vertexOutputs.vBumpUV = (uniforms.bumpMatrix *  vec4f(input.uv2, 1.0, 0.0)).xy;
```

File: packages/dev/core/src/ShadersWGSL/geometry.vertex.fx [hunk 6]
```diff
@@ -0,0 +1,184 @@
...
+			#endif
+			#ifdef REFLECTIVITY_UV2
+			vertexOutputs.vReflectivityUV = (uniforms.reflectivityMatrix *  vec4f(uv2, 1.0, 0.0)).xy;
```
Comment (noname0310) [inline packages/dev/core/src/ShadersWGSL/geometry.vertex.fx:175]:
```suggestion
			vertexOutputs.vReflectivityUV = (uniforms.reflectivityMatrix *  vec4f(input.uv2, 1.0, 0.0)).xy;
```

File: packages/dev/core/src/ShadersWGSL/geometry.vertex.fx [hunk 7]
```diff
@@ -0,0 +1,184 @@
...
+			#endif
+			#ifdef ALBEDO_UV2
+			vertexOutputs.vAlbedoUV = (uniforms.albedoMatrix *  vec4f(uv2, 1.0, 0.0)).xy;
```
Comment (noname0310) [inline packages/dev/core/src/ShadersWGSL/geometry.vertex.fx:178]:
```suggestion
			vertexOutputs.vAlbedoUV = (uniforms.albedoMatrix *  vec4f(input.uv2, 1.0, 0.0)).xy;
```

File: packages/dev/core/src/ShadersWGSL/motionBlur.fragment.fx [hunk 8]
```diff
@@ -0,0 +1,94 @@
+// Samplers
+varying input.vUV: vec2f;
```
Comment (noname0310) [inline packages/dev/core/src/ShadersWGSL/motionBlur.fragment.fx:2]:
```suggestion
varying vUV: vec2f;

```

**Review by noname0310 — COMMENTED** [review]
only suggested the section where the shader compile error occurred

**Pull request merged**